packages/calcite-components/src/components/swatch/swatch.scss

```diff
@@ -0,0 +1,122 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-swatch-corner-radius: Specifies the component's corner radius.
+ *
+ */
+
+// AUTO-GENERATED — do not modify. Changes will be overwritten.
+//
+// Internal CSS custom properties for component use only. Overwriting is not recommended.
+//
+// --calcite-internal-swatch-inset
+// --calcite-internal-swatch-size
+
+:host {
+  @apply block;
+  --calcite-internal-swatch-inset: var(--calcite-spacing-xxs);
+}
+
+:host([scale="s"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xl);
+  }
+}
+
+:host([scale="m"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxl);
+  }
+}
+
+:host([scale="l"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxxl);
+  }
+}
+
+.container {
+  @apply relative focus-base justify-center box-border  overflow-hidden;
+  font-size: var(--calcite-internal-swatch-font-size, var(--calcite-font-size));
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+
+  &:not(.non-interactive):hover {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):hover {
+    box-shadow: 0 0 0 2px var(--calcite-color-border-2);
+  }
+  &.selectable {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):focus {
+    @apply focus-normal;
+  }
+}
+
+.swatch {
+  position: absolute;
+  @apply inline-flex overflow-hidden absolute z-10;
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+}
+
+:host([selected]) .swatch {
+  inset: var(--calcite-internal-swatch-inset);
+  block-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+  inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+  min-inline-size: calc(var(--calcite-internal-swatch-size, auto) - calc(2 * var(--calcite-internal-swatch-inset)));
+}
+
+:host([selected]) .container {
+  box-shadow:
```

Comment:
should this use a shadow css var?

Reply:
Same as above - this is somewhat unique to this component, cc @ashetland

Reply:
_If_ we want to, we could use `var(--calcite-border-width-md)` and `var(--calcite-border-width-lg)` for the 2px and 4px spreads, respectively.